src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenFinalOrderDelayReason.java

```diff
@@ -51,14 +52,22 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
         log.info("Citizen final order delay reason about to submit callback invoked for Case Id: {}", details.getId());
 
         CaseData data = details.getData();
+        State endState = FinalOrderRequested;
 
         if (data.getApplicationType().isSole()) {
             notificationDispatcher.send(soleAppliedForFinalOrderNotification, data, details.getId());
         }
 
+        if (data.isWelshApplication()) {
+            data.getApplication().setWelshPreviousState(endState);
```

Comment:
why not set the state from details.getState rather than hardcoding ? We need to set the existing state right ?

Reply:
@nitinprabhu we need to set the state that it would have become if it wasn't for the Welsh translation. 
In this case if we used details.getState() we'd get FinalOrderOverdue which is wrong as we should be setting it as FinalOrderRequested (i.e the state that this event would be set to if it wasn't for Welsh)